### PR TITLE
Add optional docker support.

### PR DESCRIPTION
**Pull request by 9021007**
This PR allows users to optionally deploy the ArcOS backend with Docker. Does not affect standard deployment.

### CODE REVIEW

**Comment by ghost** [comment]
nice

**Comment by itsnotrin** [comment]
Great work :), any reason in specific for using ubuntu? Leads to a super bloated image size :( 
image

**Comment by ghost** [comment]
> Great work :), any reason in specific for using ubuntu? Leads to a super bloated image size :( image

they're probably use to ubuntu sooo

**Comment by itsnotrin** [comment]
Technically should still build on debian or alpine, I'll try building it now on alpine image :)

**Comment by ghost** [comment]
> Great work :), any reason in specific for using ubuntu? Leads to a super bloated image size :( image

wait, **1GB**?? :eyes:

**Comment by itsnotrin** [comment]
Way better now ;)

image

**Comment by IzKuipers** [comment]
> Technically should still build on debian or alpine, I'll try building it now on alpine image :)

it's literally just Python so it should run on anything that runs python 3

**Comment by ghost** [comment]
> Way better now ;)
> 
> image

thats true

**Comment by ghost** [comment]
> > Technically should still build on debian or alpine, I'll try building it now on alpine image :)
> 
> it's literally just Python so it should run on anything that runs python 3

yeah but the image size was like **1GB** so

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# start with ubuntu
+FROM python:3.11.7
```
Comment (itsnotrin) [inline Dockerfile:2]: Can this be changed out for `FROM python:3.12.1-alpine3.19`?

Comment (itsnotrin) [inline Dockerfile]: Can this be changed out for `FROM python:3.12.1-alpine3.19`?

**Comment by IzKuipers** [comment]
> > > Technically should still build on debian or alpine, I'll try building it now on alpine image :)
> 
> > 
> 
> > it's literally just Python so it should run on anything that runs python 3
> 
> 
> 
> yeah but the image size was like **1GB** so

what does that matter in response to me, swap it out for debian and it'll be smaller

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# install python3 and pip3
+RUN apt-get update && apt-get install -y python3 python3-pip python3-venv
```
Comment (itsnotrin) [inline Dockerfile]: This can be removed entirely.

**Review by itsnotrin — CHANGES_REQUESTED** [review]
Overall bloody amazing PR, just this will greatly help reduce image size.

**Comment by ghost** [comment]
> > > > Technically should still build on debian or alpine, I'll try building it now on alpine image :)
> > 
> > 
> > > 
> > 
> > 
> > > it's literally just Python so it should run on anything that runs python 3
> > 
> > 
> > yeah but the image size was like **1GB** so
> 
> what does that matter in response to me, swap it out for debian and it'll be smaller

thats what they are gonna do (just with `alpine` instead)

**Comment by itsnotrin** [comment]
Alpine results in an even smaller image than debian and is a lighter overall image in regards to resources too. Far better idea imo

File: Dockerfile (hunk 2 above)
Comment (ghost) [reply Dockerfile]: bro was installing python even though while using the `python` image :skull:

**Comment by itsnotrin** [comment]
@9021007 I'll be happy to merge this straight in once the changes are done, really amazing work man!

**Comment by itsnotrin** [comment]
Just finished testing and can confirm it fully works as normal too!

**Comment by 9021007** [comment]
Sounds good! I used the official python image, didn't check how big it was lol. Glad you got the size down.

**Comment by eversiege** [comment]
We use Debian on the server, is there any reason why Dockerfile uses Ubuntu? I applaud your effort, awesome stuff. I am not sure if I am available to push this to the main production due to undergoing internal changes. (me and SSM team working on this as we speak)

**Comment by 9021007** [comment]
I have switched to alpine, now 150MB
image

File: Dockerfile (hunk 1 above)
Comment (9021007) [reply Dockerfile]: Set to `python:3.11.7-alpine`

File: Dockerfile (hunk 2 above)
Comment (9021007) [reply Dockerfile]: Done

**Comment by itsnotrin** [comment]
Merged, thank you so much!

**Pull request merged**